Werewolf for Telegram/Werewolf Node/Werewolf.cs

```diff
@@ -4626,63 +4631,140 @@ private bool DoGameEnd(ITeam team)
                         var deathmessage = "";
                         switch (alives.Count())
                         {
+                            case 4:
+                                // Tanner and sorcerer and thief and doppelganger, tanner will die.
+                                if (alives.Any(x => x.PlayerRole == IRole.Tanner) && alives.Any(x => x.PlayerRole == IRole.Sorcerer) && alives.Any(x => x.PlayerRole == IRole.Thief) && alives.Any(x => x.PlayerRole == IRole.Doppelgänger))
+                                {
+                                    var doppelganger = alives.FirstOrDefault(x => x.PlayerRole == IRole.Doppelgänger);
+                                    var thief = alives.FirstOrDefault(x => x.PlayerRole == IRole.Thief);
+                                    var sorc = alives.FirstOrDefault(x => x.PlayerRole == IRole.Sorcerer);
+                                    var tann = alives.FirstOrDefault(x => x.PlayerRole == IRole.Tanner);
+                                    
+                                    if (doppelganger != null && sorc != null && tann != null)
+                                    {
+                                        KillPlayer(tann, KillMthd.Suicide, killer: tann, isNight: false);
+                                        
+                                        //check if the tanner is role model of doppelganger or not. If yes, also kill the doppelganger after his transform
+                                        CheckDoppelganger();
+                                        if (doppelganger.PlayerRole == IRole.Tanner)
+                                        {
+                                            KillPlayer(doppelganger, KillMthd.Suicide, killer: doppelganger, isNight: false);
+                                        }
+                                        AddAchievement(sorc, AchievementsReworked.TimeToRetire);
+                                        
+                                        deathmessage = GetLocaleString("SorcererEnd", sorc.GetName()) + Environment.NewLine;
+                                        deathmessage += Environment.NewLine + GetLocaleString("ThiefEnd", thief.GetName()) + Environment.NewLine;
+                                        deathmessage += Environment.NewLine + GetLocaleString("TannerEnd", tann.GetName()) + Environment.NewLine;
+                                        deathmessage += Environment.NewLine + GetLocaleString($"{doppelganger.PlayerRole}End", doppelganger.GetName());
+                                    }
+                                }
+                                break;
+                            case 3:
+                                // Includes three of these roles: Tanner, Thief, Doppelganger, Sorcerer
+                                var doppelganger = alives.FirstOrDefault(x => x.PlayerRole == IRole.Doppelgänger);
```

Comment:
... here, you introduce a new variable with the same name again. This won't compile unless you remove the `var` keyword. The same applies for the other variable names that you introduced multiple times.